spot-client/src/common/remote-control/xmpp-connection.js

```diff
@@ -119,8 +127,13 @@ export default class XmppConnection extends Emitter {
         } = options;
 
         if (this.initPromise) {
+            logger.error('joinMuc called before another joinMuc job completed');
+
             return this.initPromise;
         }
+        if (this._isXmppConnectionActive) {
+            logger.error('joinMuc called while another connection is still active');
```

Comment:
Maybe it should throw?

Reply:
maybe it should, I can do this change if you like... I was just hoping that the app might be still working to some extend, but would see in the logs that things are messed up

Reply:
I will change those log error to throws. We should be able to catch those during testing for the most part.